Makefile

```diff
@@ -199,6 +201,84 @@ rpm: $(RPM_SPECFILE) $(RPM_TARBALL)
 		--define "_topdir $(CURDIR)/rpmbuild" \
 		$(RPM_SPECFILE)
 
+#
+# Coverity
+#
+# Download the coverity analysis tool and run it on the repository,
+# archive the analysis result and upload it to coverity. The target
+# to do all of that is `coverity-submit`.
+#
+# Individual targets exists for the respective steps.
+#
+# Needs COVERITY_TOKEN and COVERITY_EMAIL to be set for downloading
+# the analysis tool and submitting the final results
+
+COVERITY_URL=https://scan.coverity.com/download/linux64
+COVERITY_DIR=cov-analysis-linux64
+COVERITY_TARBALL=coverity-tool.tar.gz
+COVERITY_ANALYSIS=cov-analysis-osbuild.xz
+
+.PHONY: test-coverity
+test-coverity: coverity-submit
+
+.PHONY: coverity-download
+coverity-download: $(COVERITY_TARBALL)
+
+.PHONY: coverity-check
+coverity-check: $(COVERITY_ANALYSIS)
+
+$(COVERITY_TARBALL):
+ifeq ($(COVERITY_TOKEN),)
+	@echo "COVERITY_TOKEN unset, need it."
+	@exit 1
+endif
+	@echo "Downloading $(COVERITY_TARBALL) from $(COVERITY_URL)"
+	@wget -q $(COVERITY_URL) --post-data "token=$(COVERITY_TOKEN)&project=osbuild" -O $(COVERITY_TARBALL)
+
+
+$(COVERITY_DIR): $(COVERITY_TARBALL)
+	@echo "Extracting coverity tool"
+	@mkdir $(COVERITY_DIR)
+	@tar xzf $(COVERITY_TARBALL) --strip 1 -C $(COVERITY_DIR)
+
+$(COVERITY_ANALYSIS): $(COVERITY_DIR)
+	$(PWD)/$(COVERITY_DIR)/bin/cov-build \
+		--dir cov-int \
+		--no-command \
+		--fs-capture-search ./ \
+		--fs-capture-search-exclude-regex "$(COVERITY_DIR)"
+	@echo "Compressing analysis results"
+	@tar caf $(COVERITY_ANALYSIS) cov-int
+
+coverity-submit: $(COVERITY_ANALYSIS)
+ifeq ($(COVERITY_TOKEN),)
+	@echo "COVERITY_TOKEN unset, need it."
+	@exit 1
+endif
+ifeq ($(COVERITY_EMAIL),)
+	@echo "COVERITY_EMAIL unset, need it."
+	@exit 1
+endif
+
+	DESCRIPTION=$(shell git describe)
+	@echo "Submitting $(COVERITY_ANALYSIS)"
+	@curl --form token=$(COVERITY_TOKEN) \
+	--form email=$(COVERITY_EMAIL) \
+	--form file=@$(COVERITY_ANALYSIS) \
+	--form version="main" \
+	--form description="$(DESCRIPTION)" \
+	https://scan.coverity.com/builds?project=osbuild
+
+.PHONY: coverity-clean
+coverity-clean:
+	rm -rf cov-int
+	rm -rf $(COVERITY_DIR)
+
+.PHONY: coverity-clean-all
+coverity-clean-all: coverity-clean
+	rm -rf $(COVERITY_TARBALL)
+
+
```

Comment:
I reworked the Makefile-parts a bit so they no longer use directories as dependencies (which has odd side-effects when directory timestamps change). My suggested diff is here:

https://gist.github.com/dvdhrm/51e3d24f84612b9d1560bfc291918c04

This has a few other changes as well:
  * Drop `test-coverity`. I suggest using `coverity-submit` directly rather than an alias.
  * Use `$(error ...)` with order-only targets to pull in TOKEN and EMAIL. This avoids the conditionals.
  * Honor `$(BUILDDIR)` and `$(SRCDIR)`. This allows doing `make coverity-check BUILDDIR=../build` which then does not pollute the srcdir with artifacts. Or, assuming your source is in `./osbuild`, you can now do: `make -C osbuild BUILDDIR=../build coverity-check`
  * Make `coverity-download` forcibly download the sources, so it can be used to re-download everything. However, make sure this is not triggered by `coverity-check`, unless the files do not exist, yet. This also means `TOKEN` needs not be set for `coverity-check`, unless the files have not been downloaded, yet.
  * Some minor formatting fixes to align with coding-style of Makefile.

Feel free to cherry-pick what you like, or to adjust to your needs.